lib/src/merged_tree.rs

```diff
@@ -146,19 +156,26 @@ impl MergedTree {
     /// Async version of `path_value()`.
     pub async fn path_value_async(&self, path: &RepoPath) -> BackendResult<MergedTreeValue> {
         match path.split() {
-            Some((dir, basename)) => match self.trees.sub_tree_recursive(dir).await? {
-                None => Ok(Merge::absent()),
-                Some(tree) => Ok(tree.value(basename).cloned()),
-            },
-            None => Ok(self
-                .trees
-                .map(|tree| Some(TreeValue::Tree(tree.id().clone())))),
+            Some((dir, basename)) => {
+                let trees = self.trees_async().await?;
+                match trees.sub_tree_recursive(dir).await? {
+                    None => Ok(Merge::absent()),
+                    Some(tree) => Ok(tree.value(basename).cloned()),
+                }
+            }
+            None => Ok(self.as_merged_tree_value()),
         }
     }
 
+    fn as_merged_tree_value(&self) -> MergedTreeValue {
```

Comment:
nit: I think this should be called `to_*` by convention since it doesn't just return a reference